rule/confusing-naming.go

```diff
@@ -0,0 +1,102 @@
+package rule
+
+import (
+	"fmt"
+	"go/ast"
+	"strings"
+
+	"github.com/mgechev/revive/lint"
+)
+
+// ConfusingNamingRule lints method names that differ only by capitalization
+type ConfusingNamingRule struct{}
+
+// Apply applies the rule to given file.
+func (r *ConfusingNamingRule) Apply(file *lint.File, arguments lint.Arguments) []lint.Failure {
+	var failures []lint.Failure
+
+	fileAst := file.AST
+	walker := lintConfusingNames{
+		methodNames: make(map[string][]*ast.Ident),
```

Comment:
If we change the type of `methodNames` to `map[string]map[string]bool` where the key in the nested map is lower/uppercase identifier, we can make the `checkMethodName` run in `O(1)`.

Reply:
done